kcalcore/icaltimezones.cpp

```diff
@@ -1313,10 +1313,10 @@ QList<QDateTime> ICalTimeZoneSourcePrivate::parsePhase( icalcomponent *c,
   }
 
   // Convert DTSTART to QDateTime, and from local time to UTC
-  const QDateTime localStart = toQDateTime( dtstart );   // local time
   dtstart.second -= prevOffset;
   dtstart.is_utc = 1;
   const QDateTime utcStart = toQDateTime( icaltime_normalize( dtstart ) );   // UTC
+  const QDateTime localStart = utcStart.toLocalTime();  // local time
```

Comment:
I built the new libical, installed it to SDK, then rebuilt kcalcore from master and install that to SDK, then rebuilt my test app, and deployed all of the packages to device -> after running test app, results are unchanged, all three tests fail.  I then rebuilt kcalcore with just the second patch applied (ie, the change to appendDateTime) on top of the new libical, and deployed that to the device, then running the test app showed all succeed (but still the interval lower-bound is printed out as empty).

In short, it looks like the new libical doesn't affect the results; the bugs are in kcalcore, I think, not libical.

I've attempted three times now to contact the kcalcore developers (first by kde-pim mailinglist, then twice by IRC on both #kde and #kontact) to ask for their comments on this PR but have received no response.  I agree that the second patch especially seems strange, but this bug is critical and really frustrating our users...